src/engine/renderer/tr_local.h

```diff
@@ -3183,15 +3183,15 @@ inline bool checkGLErrors()
 	template<size_t frustumSize>
 	inline
 	typename std::enable_if<(frustumSize >= frustumBits_t::FRUSTUM_NEAR + 1)>::type
-	R_CalcFrustumNearCorners( const vec4_t(&frustum)[frustumSize], vec3_t (&corners)[ 4 ] ) {
-		extern void R_CalcFrustumNearCornersUnsafe( const vec4_t frustum[frustumBits_t::FRUSTUM_NEAR + 1], vec3_t (&corners)[ 4 ] );
+	R_CalcFrustumNearCorners( const plane_t(&frustum)[frustumSize], vec3_t (&corners)[ 4 ] ) {
```

Comment:
A frustum is not a plane. plane_t is not appropriate here

Reply:
I don't know what is a frustum, but if frustum is not a plane or doesn't have a plane, maybe we need another name for a struct composed of a normal vec3 and a distance float.

This code was already splitting the data into a normal vec3 and a distance float and naming it plane.

Examples:

```c++
VectorCopy(plane, tr.viewParms.frustums[0][i].normal);
tr.viewParms.frustums[0][i].dist = plane[3];
```

```c++
VectorNegate( lightProject[ 3 ], frustum[ FRUSTUM_FAR ] );
frustum[ FRUSTUM_FAR ][ 3 ] = -lightProject[ 3 ][ 3 ] - 1.0f;
```

```c++
using frustum_t = cplane_t[6];
```

```c++
struct cplane_t
{
	vec3_t normal;
	float  dist;
	byte   type; // for fast side tests: 0,1,2 = axial, 3 = nonaxial
	byte   signbits; // signx + (signy<<1) + (signz<<2), used as lookup during collision
	byte   pad[ 2 ];
};
```

All of this is code from _before_ my pull request.

Reply:
@VReaperV how would you name a generic struct that would be composed like this:

```c++
struct something_t {
	float normal[3];
	float distance;
};
```

Reply:
Here I read in French:

> On appelle espace métrique tout couple (E, d) o`u E 6 ≠ ∅ est un espace vectoriel et d est une distance.
> -- https://licence-math.univ-lyon1.fr/lib/exe/fetch.php?media=pmi:td3_-_topologie.pdf

Wikipedia says:

> Formally, a metric space is an [ordered pair](https://en.wikipedia.org/wiki/Ordered_pair) (M, d) where M is a set and d is a metric on M, i.e., a [function](https://en.wikipedia.org/wiki/Function_(mathematics))
>  -- https://en.wikipedia.org/wiki/Metric_space

With “a set” just being defined as:

> A set is the [mathematical model](https://en.wikipedia.org/wiki/Mathematical_model) for a collection of different things;
> -- https://en.wikipedia.org/wiki/Set_(mathematics)

Without normal meaning.

Reply:
Frustum is a set of planes. More specifically the 6 planes that are the near and far clipping planes, and 4 planes that intersect the view origin, i. e. a pyramid with its top cut off (for the case of frustums used in rendering).

Reply:
This `const vec4_t(&frustum)[frustumSize]` seems to just be getting the pointer to the first member of a plane in `frustum`, i. e. `cplane_t.normal[0]`. So that change looks correct to me since only normal and dist seem to be used.